Block/BlockContextManagerInterface.php

```diff
@@ -39,6 +39,13 @@ public function addSettingsByType($type, array $settings, $replace = false);
     public function addSettingsByClass($class, array $settings, $replace = false);
 
     /**
+     * @param string $type    Block type to check for
+     *
+     * @return bool
+     */
+    public function exists($type);
```

Comment:
adding a method to an interface is a BC break, and according to semver, this is not OK within minor versions. if i wrote my own implementation of this interface and update a minor version, i would suddenly see errors.

the BC way of doing this would be an additional interface that extends this interface to add the exists method, and the code that wants to do the exists checks would check if that additional interface is present. in the next major version, you could merge the two interfaces and simplify the code.